--- lttng-modules-2.5.1/Makefile~	2014-10-20 22:48:47.000000000 +0200
+++ lttng-modules-2.5.1/Makefile	2014-11-09 16:11:25.914717733 +0100
@@ -5,19 +5,19 @@
 ifneq ($(KERNELRELEASE),)
 ifneq ($(CONFIG_TRACEPOINTS),)
 
-lttng_check_linux_version = $(shell pwd)/include/linux/version.h
-lttng_check_generated_linux_version = $(shell pwd)/include/generated/uapi/linux/version.h
-
+#lttng_check_linux_version = $(shell pwd)/include/linux/version.h
+#lttng_check_generated_linux_version = $(shell pwd)/include/generated/uapi/linux/version.h
+#
 #
 # Check for stale version.h, which can be a leftover from an old Linux
 # kernel tree moved to a newer kernel version, only pruned by make
 # distclean.
 #
-ifneq ($(wildcard $(lttng_check_linux_version)),)
-ifneq ($(wildcard $(lttng_check_generated_linux_version)),)
-$(error Duplicate version.h files found in $(lttng_check_linux_version) and $(lttng_check_generated_linux_version). Consider running make distclean on your kernel, or removing the stale $(lttng_check_linux_version) file)
-endif
-endif
+#ifneq ($(wildcard $(lttng_check_linux_version)),)
+#ifneq ($(wildcard $(lttng_check_generated_linux_version)),)
+#$(error Duplicate version.h files found in $(lttng_check_linux_version) and $(lttng_check_generated_linux_version). Consider running make distclean on your kernel, or removing the stale $(lttng_check_linux_version) file)
+#endif
+#endif
 
 obj-m += lttng-ring-buffer-client-discard.o
 obj-m += lttng-ring-buffer-client-overwrite.o
