diff --git a/include/instrumentation/events/jbd2.h b/include/instrumentation/events/jbd2.h
index d5d8ea0c..be217ffa 100644
--- a/include/instrumentation/events/jbd2.h
+++ b/include/instrumentation/events/jbd2.h
@@ -28,6 +28,8 @@ LTTNG_TRACEPOINT_EVENT(jbd2_checkpoint,
 )
 
 #if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(6,2,0) \
+	|| LTTNG_KERNEL_RANGE(5,4,229, 5,5,0) \
+	|| LTTNG_KERNEL_RANGE(5,10,163, 5,11,0) \
 	|| LTTNG_KERNEL_RANGE(5,15,87, 5,16,0) \
 	|| LTTNG_KERNEL_RANGE(6,0,18, 6,1,0) \
 	|| LTTNG_KERNEL_RANGE(6,1,4, 6,2,0))
@@ -96,6 +98,8 @@ LTTNG_TRACEPOINT_EVENT_INSTANCE(jbd2_commit, jbd2_drop_transaction,
 #endif
 
 #if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(6,2,0) \
+	|| LTTNG_KERNEL_RANGE(5,4,229, 5,5,0) \
+	|| LTTNG_KERNEL_RANGE(5,10,163, 5,11,0) \
 	|| LTTNG_KERNEL_RANGE(5,15,87, 5,16,0) \
 	|| LTTNG_KERNEL_RANGE(6,0,18, 6,1,0) \
 	|| LTTNG_KERNEL_RANGE(6,1,4, 6,2,0))
@@ -138,6 +142,8 @@ LTTNG_TRACEPOINT_EVENT(jbd2_submit_inode_data,
 )
 
 #if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(6,2,0) \
+	|| LTTNG_KERNEL_RANGE(5,4,229, 5,5,0) \
+	|| LTTNG_KERNEL_RANGE(5,10,163, 5,11,0) \
 	|| LTTNG_KERNEL_RANGE(5,15,87, 5,16,0) \
 	|| LTTNG_KERNEL_RANGE(6,0,18, 6,1,0) \
 	|| LTTNG_KERNEL_RANGE(6,1,4, 6,2,0))
