--- lttng-modules-2.11.1/instrumentation/events/lttng-module/arch/x86/kvm/mmutrace.h~	2020-01-28 16:09:46.000000000 +0100
+++ lttng-modules-2.11.1/instrumentation/events/lttng-module/arch/x86/kvm/mmutrace.h	2020-02-23 07:58:12.916686709 +0100
@@ -219,6 +219,21 @@
 
 	kvm_mmu_fast_page_fault,
 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(4,14))
+	TP_PROTO(struct kvm_vcpu *vcpu, gpa_t cr2_or_gpa, u32 error_code,
+		 u64 *sptep, u64 old_spte, bool retry),
+	TP_ARGS(vcpu, cr2_or_gpa, error_code, sptep, old_spte, retry),
+
+	TP_FIELDS(
+		ctf_integer(int, vcpu_id, vcpu->vcpu_id)
+		ctf_integer(gpa_t, cr2_or_gpa, cr2_or_gpa)
+		ctf_integer(u32, error_code, error_code)
+		ctf_integer_hex(u64 *, sptep, sptep)
+		ctf_integer(u64, old_spte, old_spte)
+		ctf_integer(u64, new_spte, *sptep)
+		ctf_integer(bool, retry, retry)
+	)
+#else
 	TP_PROTO(struct kvm_vcpu *vcpu, gva_t gva, u32 error_code,
 		 u64 *sptep, u64 old_spte, bool retry),
 	TP_ARGS(vcpu, gva, error_code, sptep, old_spte, retry),
@@ -232,6 +247,7 @@
 		ctf_integer(u64, new_spte, *sptep)
 		ctf_integer(bool, retry, retry)
 	)
+#endif
 )
 #endif /* LTTNG_TRACE_KVM_MMU_H */
 
