diff -ur lttng-modules-2.13.11/lttng-modules-2.13.11/src/lib/ringbuffer/ring_buffer_backend.c lttng-modules-2.13.11.strcpy/lttng-modules-2.13.11/src/lib/ringbuffer/ring_buffer_backend.c
--- lttng-modules-2.13.11/lttng-modules-2.13.11/src/lib/ringbuffer/ring_buffer_backend.c	2024-01-10 21:35:48.000000000 +0100
+++ lttng-modules-2.13.11.strcpy/lttng-modules-2.13.11/src/lib/ringbuffer/ring_buffer_backend.c	2024-03-17 00:20:50.475514756 +0100
@@ -405,7 +405,7 @@
 	chanb->extra_reader_sb =
 			(config->mode == RING_BUFFER_OVERWRITE) ? 1 : 0;
 	chanb->num_subbuf = num_subbuf;
-	strlcpy(chanb->name, name, NAME_MAX);
+	strscpy(chanb->name, name, NAME_MAX);
 	memcpy(&chanb->config, config, sizeof(chanb->config));
 
 	if (config->alloc == RING_BUFFER_ALLOC_PER_CPU) {
