From 33630522da97345c2b0c569c04c1d5d33ab7efa6 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 26 Dec 2017 09:47:22 -0500
Subject: [PATCH] Update kvm instrumentation for 4.15

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index a8b3e9a..c01772c 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -84,6 +84,22 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 	{ KVM_TRACE_MMIO_READ, "read" }, \
 	{ KVM_TRACE_MMIO_WRITE, "write" }
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+
+LTTNG_TRACEPOINT_EVENT(kvm_mmio,
+	TP_PROTO(int type, int len, u64 gpa, void *val),
+	TP_ARGS(type, len, gpa, val),
+
+	TP_FIELDS(
+		ctf_integer(u32, type, type)
+		ctf_integer(u32, len, len)
+		ctf_integer(u64, gpa, gpa)
+		ctf_sequence_hex(unsigned char, val, val, u32, len)
+	)
+)
+
+#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	TP_PROTO(int type, int len, u64 gpa, u64 val),
 	TP_ARGS(type, len, gpa, val),
@@ -96,6 +112,8 @@ LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	)
 )
 
+#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,34))
 
 #define kvm_fpu_load_symbol	\
From 5ccd696806d409d5d0261b2e7e4d9618daf02e24 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 2 Jan 2018 11:07:05 -0500
Subject: [PATCH] Update: kvm instrumentation for 3.16.52 and 3.2.97

Starting from 3.16.52 and 3.2.97, the 3.16 and 3.2 stable kernel
branches backport a kvm instrumentation change introduced in 4.15 which
affects the prototype of the kvm_mmio event.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index ec74bdd..ea63e88 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -84,7 +84,9 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 	{ KVM_TRACE_MMIO_READ, "read" }, \
 	{ KVM_TRACE_MMIO_WRITE, "write" }
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0) \
+	|| LTTNG_KERNEL_RANGE(3,16,52, 3,17,0) \
+	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0))
 
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	TP_PROTO(int type, int len, u64 gpa, void *val),
@@ -98,7 +100,7 @@ LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	)
 )
 
-#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+#else
 
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	TP_PROTO(int type, int len, u64 gpa, u64 val),
@@ -112,7 +114,7 @@ LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	)
 )
 
-#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+#endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,34))
 
From bda29b79b33c2a31d3ffb396e741724590f6e455 Mon Sep 17 00:00:00 2001
From: Michael Jeanson <mjeanson@efficios.com>
Date: Tue, 9 Jan 2018 15:43:20 -0500
Subject: [PATCH] Update kvm instrumentation for debian kernel 4.9.65-3

Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index ea63e88..d9d0300 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -86,7 +86,8 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0) \
 	|| LTTNG_KERNEL_RANGE(3,16,52, 3,17,0) \
-	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0))
+	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0) \
+	|| LTTNG_DEBIAN_KERNEL_RANGE(4,9,65,0,3,0, 4,10,0,0,0,0))
 
 LTTNG_TRACEPOINT_EVENT(kvm_mmio,
 	TP_PROTO(int type, int len, u64 gpa, void *val),
From 1d822c67df88b678c9cbe73d0bb47fc177af194a Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Wed, 17 Jan 2018 11:17:08 -0500
Subject: [PATCH] Update: kvm instrumentation for 4.14.14+, 4.9.77+, 4.4.112+

Starting from 3.14.14, 4.9.77, and 4.4.112, the 3.14, 4.9, and 4.4
stable kernel branches backport a kvm instrumentation change introduced
in 4.15 which affects the prototype of the kvm_mmio event.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index d9d0300..fbd603b 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -85,6 +85,9 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 	{ KVM_TRACE_MMIO_WRITE, "write" }
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0) \
+	|| LTTNG_KERNEL_RANGE(4,14,14, 4,15,0) \
+	|| LTTNG_KERNEL_RANGE(4,9,77, 4,10,0) \
+	|| LTTNG_KERNEL_RANGE(4,4,112, 4,5,0) \
 	|| LTTNG_KERNEL_RANGE(3,16,52, 3,17,0) \
 	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0) \
 	|| LTTNG_DEBIAN_KERNEL_RANGE(4,9,65,0,3,0, 4,10,0,0,0,0))
