--- /home/users/baggins/devel/PLD/rpm/BUILD/lttng-modules-2.5.1/instrumentation/events/lttng-module/compaction.h~	2014-10-20 22:48:47.000000000 +0200
+++ /home/users/baggins/devel/PLD/rpm/BUILD/lttng-modules-2.5.1/instrumentation/events/lttng-module/compaction.h	2014-11-23 15:28:33.501139643 +0100
@@ -46,7 +46,7 @@
 	TP_ARGS(nr_scanned, nr_taken)
 )
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,16,0)) || ((LINUX_VERSION_CODE >= KERNEL_VERSION(3,14,25)) && (LINUX_VERSION_CODE < KERNEL_VERSION(3,15,0)))
 TRACE_EVENT(mm_compaction_migratepages,
 
 	TP_PROTO(unsigned long nr_all,
