From f279893a432a5c5c5d4488f65a0719181a06da56 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Tue, 26 Dec 2017 09:47:36 -0500
Subject: [PATCH] Update sock instrumentation for 4.15

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/sock.h | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/instrumentation/events/lttng-module/sock.h b/instrumentation/events/lttng-module/sock.h
index e79f8dc..5cd02ca 100644
--- a/instrumentation/events/lttng-module/sock.h
+++ b/instrumentation/events/lttng-module/sock.h
@@ -5,6 +5,7 @@
 #define LTTNG_TRACE_SOCK_H
 
 #include <probes/lttng-tracepoint-event.h>
+#include <linux/version.h>
 #include <net/sock.h>
 
 LTTNG_TRACEPOINT_EVENT(sock_rcvqueue_full,
@@ -20,6 +21,25 @@ LTTNG_TRACEPOINT_EVENT(sock_rcvqueue_full,
 	)
 )
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0))
+
+LTTNG_TRACEPOINT_EVENT(sock_exceed_buf_limit,
+
+	TP_PROTO(struct sock *sk, struct proto *prot, long allocated),
+
+	TP_ARGS(sk, prot, allocated),
+
+	TP_FIELDS(
+		ctf_string(name, prot->name)
+		ctf_array(long, sysctl_mem, prot->sysctl_mem, 3)
+		ctf_integer(long, allocated, allocated)
+		ctf_integer(int, sysctl_rmem, sk_get_rmem0(sk, prot))
+		ctf_integer(int, rmem_alloc, atomic_read(&sk->sk_rmem_alloc))
+	)
+)
+
+#else /* #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 LTTNG_TRACEPOINT_EVENT(sock_exceed_buf_limit,
 
 	TP_PROTO(struct sock *sk, struct proto *prot, long allocated),
@@ -35,6 +55,8 @@ LTTNG_TRACEPOINT_EVENT(sock_exceed_buf_limit,
 	)
 )
 
+#endif /* #else #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)) */
+
 #endif /* LTTNG_TRACE_SOCK_H */
 
 /* This part must be outside protection */
